Month1/Week2/Section2/Month1-Week2-Section2/Spotify/PlayList.cpp

```diff
@@ -0,0 +1,52 @@
+#include"PlayList.h"
+
+FPlayList::FPlayList()
+{
+	PlayListTitle = "";
+	SongCount = 0;
+}
+FPlayList::FPlayList(std::string NewTitle)
+{
+	PlayListTitle = NewTitle;
+}
+
+std::string FPlayList::GetListTitle()
+{
+	return PlayListTitle;
+}
+
+void FPlayList::SetListTitle(std::string NewPlayListTitle)
+{
+	PlayListTitle = NewPlayListTitle;
+}
+FSong FPlayList::GetSong(int SongIndex)
+{
+	return Songs[SongIndex];
+}
+bool FPlayList::bAddSong(FSong Song)
+{
+	if (Song.bIsvalid())
+	{
+		Songs[SongCount] = Song;
+		SongCount++;
+		return true;
+	}
+	return false;
+}
+
+
+void  FPlayList::PlayListDisplay()
+{
+	std::cout << "PlayList" << std::endl;
+	std::cout << "Song Title\t\t\t\tArtist Name\t\t\t\tDuration" << std::endl;
+	std::cout << "------------------------------------------------------------------------------------------" << std::endl;
+	for (int CurrentSong = 0; CurrentSong < SongCount; CurrentSong++)
```

Comment:
This is an example of misleading names. I know you like to name your indices explicitly and that's actually very nice. However, keep in mind that they are still _indices_. `CurrentSong` is not actually the current song, but rather the current song _index_. By itself, this can be misleading, but it becomes a bigger problem since now you have used that name, and so you can't use it for anything else.

For that reason, when you actually get the song using the index, you can't name it `CurrentSong` so you had to name it `ListedSong`. Now, by just reading the names, would you know what the difference between `CurrentSong` and `ListedSong` is? If it was me, I'd assume that `CurrentSong` is the actual current song and I would have no idea what `ListedSong` is, which is not good 😅 .

It's cases like this one that show why being explicit is important. `CurrentSong` should instead be `CurrentSongIndex` or something like that, and `ListedSong` should be `CurrentSong`. What do you think?

Reply:
It's reasonable, thank you, changed